packages/relay/src/lib/eth.ts

```diff
@@ -669,12 +669,26 @@ export class EthImpl implements Eth {
                   currentTimestamp = mirrorAccount.balance.timestamp;
                 }
 
-                let transactionsInTimeWindow = await this.mirrorNodeClient.getTransactionsForAccount(
-                  mirrorAccount.account,
-                  block.timestamp.to,
-                  currentTimestamp,
-                  requestId
-                );
+                // Need to check if there are any transactions before the block.timestamp.to in the current account set returned from the inital
+                // call to getAccountPageLimit.  If there are we may need to paginate.
+                let lastTransactionOnPageTimestamp;
+                if(mirrorAccount.links.next !== null) {
+                  // Get the end of the page of transactions timestamp
+                  lastTransactionOnPageTimestamp = mirrorAccount.links.next.substring(mirrorAccount.links.next.indexOf("gte:") + 4, mirrorAccount.links.next.indexOf("&timestamp"));
+                }
+                let transactionsInTimeWindow: any = [];
+                if((typeof lastTransactionOnPageTimestamp !== "undefined") && (mirrorAccount.transactions[mirrorAccount.transactions.length -1].consensus_timestamp >= lastTransactionOnPageTimestamp)) {
```

Comment:
Q: what's this testing or looking for and what's the significance?
Also unclear how they connect to the `block.timestamp.to` and `currentTimestamp` boundaries when true

Reply:
No longer necessary with the improved parsing described above.  It was trying to avoid a null or undefined related error before testing the value.

Reply:
The code is still there. I'm not clear here what this if is doing.
Are you just checking to see if there are more transaction since the next link has a value?